test/specs/scenarios/crossDevice.js

```diff
@@ -421,6 +421,46 @@ export const crossDeviceScenarios = async (lang) => {
         crossDeviceSubmit.clickOnSubmitVerificationButton()
         verificationComplete.checkLogoIsHidden()
       })
+
+      it('should continue through full flow without problems when decoupled but still uploading media', async () => {
+        driver.get(
+          `${baseUrl}&useDecoupleCallbacks=true&decoupleResponse=onfido`
+        )
+        welcome.continueToNextStep()
+        documentSelector.clickOnPassportIcon()
+        runThroughCrossDeviceFlow()
+        documentUpload.clickUploadButton()
+        uploadFileAndClickConfirmButton(
+          passportUploadImageGuide,
+          confirm,
+          'passport.jpg'
+        )
+        selfieIntro.clickOnContinueButton()
+        camera.takeSelfie()
+        confirm.clickConfirmButton()
+        switchBrowserTab(0)
+        crossDeviceSubmit.clickOnSubmitVerificationButton()
+      })
```

Comment:
This is to test when the callback is triggered but the response indicates that the integrator still wants the SDK to upload. This would be the use case for customers who need to save the media directly from the SDK rather than getting it later from our API but don't have a problem with our SDK handling the upload once that's done.

I couldn't find a way to actually test if the API request is made though so it's mostly testing that the response is properly handled and the user doesn't get stuck. Would be happy to improve the test if you know of a way to do it.

Reply:
Ah, ok. Then maybe just add on to the description for clarity: `'should continue through full flow without problems when decoupled but still uploading media to API/as normal'`

For testing if the API request is actually made, sounds like that's a scenario that an integration test would help with. @ethan605 any thoughts on this?

Reply:
@kopijunkie doesn't sound like integration tests for me. In this case, I prefer to write unit tests for the callbacks, assume different cases the response could be, and assert if our implementation meets expectation. Technically we can't/shouldn't care about what the real customised API requests do. That's the integrator's part.

Reply:
Thanks for the suggestion. I've managed to add some unit tests for the callbacks. Let me know what you think or if I've missed something.